### PR TITLE
da: use celestia as da

### PR DESCRIPTION
**Pull request by tuxcanfly**
## Overview

This PR refactors the batcher to submit only data pointers to the batch inbox, the data
being made available on a off chain data layer, in this case celestia.

The serialization format of the data pointer is:

```
--------------------
| 8 bytes | 4 bytes |
--------------------
| height  | index   |
--------------------
```
where height is the block height at which the actual data was included and
index is the index at which the transaction was included in the block.

(TODO: clarify if we need tx index or share index or both here)

Since the batch calldata is now just a data pointer instead of the actual data, corresponding
refactor has been made to the op-node to read the data pointer off the batch inbox address first
and then retreive the data from the data layer.

This makes the rollup more modular as it makes the data available on it's own layer while using
ethereum for settlement, thereby inheriting it's security. Celestia is the data availability
layer of choice as it provides a high guarantee of the data availability.

## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

File: ops-bedrock/docker-compose.yml [hunk 1]
```diff
@@ -11,6 +11,14 @@ volumes:
 
 
 services:
+  da:
+    container_name: da
+    platform: linux/x86_64
+    image: "ghcr.io/celestiaorg/local-celestia-devnet:main"
```
Comment (jcstein) [inline ops-bedrock/docker-compose.yml]: should we lock this on a version? i.e. `v0.9.1` which is latest?

Comment (jcstein) [reply ops-bedrock/docker-compose.yml]: we also need to change this to `rollkit/local...`

Comment (jcstein) [inline ops-bedrock/docker-compose.yml]:
```suggestion
    image: "ghcr.io/rollkit/local-celestia-devnet:v0.9.1"
```

File: op-node/rollup/derive/calldata_source.go [hunk 2]
```diff
@@ -120,7 +124,36 @@ func DataFromEVMTransactions(config *rollup.Config, batcherAddr common.Address,
...
+			}
+			out = append(out, data[index])
 		}
 	}
 	return out
 }
+
+// decodeETHData will decode the data retrieved from the EVM, this data
+// was previously posted from op-batcher and contains the block height
+// with transaction index of the SubmitPFD transaction to the DA.
```
Comment (jcstein) [inline op-node/rollup/derive/calldata_source.go:145]:
```suggestion
// with transaction index of the SubmitPFB transaction to the DA.
```

**Review by jcstein — CHANGES_REQUESTED** [review]

**Comment by jcstein** [comment]
Hey @tuxcanfly can we use latest, so v0.9.1?

File: ops-bedrock/docker-compose.yml [hunk 3]
```diff
@@ -11,6 +11,14 @@ volumes:
 
 
 services:
+  da:
+    container_name: da
+    platform: linux/x86_64
+    image: "ghcr.io/rollkit/local-celestia-devnet:v0.8.2"
```
Comment (jcstein) [inline ops-bedrock/docker-compose.yml:17]:
```suggestion
    image: "ghcr.io/rollkit/local-celestia-devnet:v0.9.1"
```

File: ops-bedrock/docker-compose.yml (hunk 1 above)
Comment (tuxcanfly) [reply ops-bedrock/docker-compose.yml]: Done.

Comment (tuxcanfly) [reply ops-bedrock/docker-compose.yml]: Done.

File: op-node/rollup/derive/calldata_source.go (hunk 2 above)
Comment (tuxcanfly) [reply op-node/rollup/derive/calldata_source.go:145]: Fixed.

**Comment by tuxcanfly** [comment]
Going to merge this into `celestia-develop` branch which will track all celestia changes from now on. New PRs should be created on top of and merged into this branch for now.

**Pull request merged**